src/emotion-tracking/emotion-tracking.service.ts

```diff
@@ -0,0 +1,130 @@
+        } 
+        throw new HttpException('Emotion record of this elderly has already been made today', HttpStatus.BAD_REQUEST);
+    }
+
+    async getEmotionalRecord(caretakerId: number, elderlyId: number, limit: number, offset:number=0): Promise<EmotionHistory>{
+        await this.userService.checkRelationship(elderlyId, caretakerId);
+        let emotionalRecord;
+        let numRecords;
+        emotionalRecord = await this.emotionRecordRepository
+            .createQueryBuilder("emotionRecord")
+            .where("emotionRecord.uid = :uid", {uid: elderlyId})
+            .orderBy("emotionRecord.date", "DESC")
+            .getMany();
+        numRecords = emotionalRecord.length;
+    
+        //check if the limit is specified. default: return all
+        if(limit){
+            emotionalRecord = await this.emotionRecordRepository
+                .createQueryBuilder("emotionRecord")
+                .where("emotionRecord.uid = :uid", {uid: elderlyId})
+                .orderBy("emotionRecord.date", "DESC")
+                .limit(limit)
+                .offset(offset)
+                .getMany();
+        }
+        emotionalRecord.forEach(record => {
+            record.emotionalLevel = Emotion[record.emotionalLevel]
+        })
+        const emotionalHistory = {
+            count: numRecords,
+            records: emotionalRecord
+        }
+        return emotionalHistory;
+    }
+
+    async getEmotionTrackingStatus(caretakerId: number, elderlyId: number): Promise<EmotionTrackingStatusDto>{
+        await this.userService.checkRelationship(elderlyId, caretakerId);
+        const elderly = await this.userRepository.findOne({uid: elderlyId},{
+            relations: ["modules"]
+        })
+        const moduleFive = elderly.modules.filter(module=> module.moduleid === 5)[0];
+        let emotionTrackingStatus = new EmotionTrackingStatusDto();
+        if(moduleFive){
+            emotionTrackingStatus['isEnabled'] = true;
+        } else{
+            emotionTrackingStatus['isEnabled'] = false;
+        }
+        return emotionTrackingStatus;
+    }
+
+    async addEmotionalTrackingModuleToElderly(caretakerId: number, elderlyId: number): Promise<{statusCode: number, message: string}>{
+        await this.userService.checkRelationship(elderlyId, caretakerId);
+        const elderly = await this.userRepository.findOne({uid: elderlyId},{
+            relations: ["modules"]
+        })
+        const moduleFive = elderly.modules.filter(module=> module.moduleid === 5)[0];  //5 is the moduleid of Emotion Tracking module
+        if(moduleFive){
+            throw new HttpException("Emotion tracking is already enabled", HttpStatus.CONFLICT);
+        }
+        const module = await this.moduleRepository.findOne({moduleid: 5});
+        elderly.modules.push(module);
+        await this.userRepository.save(elderly);
+        return {statusCode: 201, message: 'Emotion tracking is successfully enabled'}
```

Comment:
Why specific status code here?

Reply:
because Jo requests it

Reply:
I think you can specify status code in the controller with `@HttpCode(201)` unless Jo wants this exact data returned.

Reply:
Any will do :)

Reply:
@chuwong35122 only the message?